lib/ansible/modules/identity/cyberark/cyberark_user.py

```diff
@@ -0,0 +1,461 @@
+
+DOCUMENTATION = '''
+---
+module: cyberark_user
+short_description: Module for CyberArk User Management using PAS Web Services SDK
+author: Edward Nunez @ CyberArk BizDev (@enunez-cyberark, @cyberark-bizdev, @erasmix)
+version_added: 2.4
+description:
+    - CyberArk User Management using PAS Web Services SDK. It currently supports the following
+      actions Get User Details, Add User, Update User, Delete User.
+
+
+options:
+    username:
+        required: True
+        description:
+            - The name of the user who will be queried (for details), added, updated or deleted.
+    state:
+        default: details
+        choices: [present, absent]
+        description:
+            - Specifies the state needed for the user
+              present for create user, absent for delete user.
+    cyberark_session:
+        required: True
+        description:
+            - Dictionary set by a CyberArk authentication containing the different values to perform actions on a logged-on CyberArk session.
+    initial_password:
+        description:
+            - The password that the new user will use to log on the first time. This password must meet the password policy requirements.
+              this parameter is required when state is present -- Add User.
+    new_password:
+        description:
+            - The user updated password. Make sure that this password meets the password policy requirements.
+    email:
+        description:
+            - The user email address.
+    first_name:
+        description:
+            - The user first name.
+    last_name:
+        description:
+            - The user last name.
+    change_password_on_the_next_logon:
+        type: bool
+        default: 'no'
+        description:
+            - Whether or not the user must change their password in their next logon.
+              Valid values = true/false.
+    expiry_date:
+        description:
+            - The date and time when the user account will expire and become disabled.
+    user_type_name:
+        default: EPVUser
+        description:
+            - The type of user.
+    disabled:
+        type: bool
+        default: 'no'
+        description:
+            - Whether or not the user will be disabled. Valid values = true/false.
+    location:
+        description:
+            - The Vault Location for the user.
+    group_name:
+        description:
+            - The name of the group the user will be added to.
+'''
+
+EXAMPLES = '''
+- name: Logon to CyberArk Vault using PAS Web Services SDK
+  cyberark_authentication:
+    api_base_url: "https://components.cyberark.local"
+    use_shared_logon_authentication: true
+
+- name: Create user & immediately add it to a group
+  cyberark_user:
+    username: "username"
+    initial_password: "password"
+    user_type_name: "EPVUser"
+    change_password_on_the_next_logon: false
+    group_name: "GroupOfUsers"
+    state: present
+    cyberark_session: "{{ cyberark_session }}"
```

Comment:
There should be an example of ```cyberark_session``` somewhere.  So that users know what to put in it.